File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def solution(left, right):
    # result = 0;
    # for i in range(left, right + 1):
    #     count = 1;
    #     for j in range (2, i + 1):
    #         if i % j == 0: count += 1;
    #     if count % 2 == 0: result += i;
    #     else: result -= i;
    # return result
    
    def calcDivisor(n):
        count = 0;
        for i in range(1, n + 1):
            if n % i == 0: count += 1;
        return n if count % 2 == 0 else -n
    
    return sum(list(map(calcDivisor, [x for x in range(left, right + 1)])))

File: 프로그래머스/lv1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.js
// function boolDivisor(n) {
//     let count = 0;
//     for(let i = 1; i <= n; i++){
//         if(n % i === 0) count++
//     }
//     return count % 2 == 0 ? 1 : -1
// }

function solution(left, right) {
    // const rangeNum = [];
    // for(let i = left; i <= right; i++){
    //     rangeNum.push(i)
    // }
    // return rangeNum.reduce((acc, cur) => acc + (cur * boolDivisor(cur)), 0)
    
    let result = 0;
    for(let i = left; i <= right; i++){
        let count = 0;
        for(let j = 1; j <= i; j++){
            if(i % j === 0) count++
        }
        if(count % 2 === 0) result += i
        else result -= i
    }
    return result
}

// 제곱근이 정수면 약수의 갯수가 홀수이다.
// function solution(left, right) {
//     var answer = 0;
//     for (let i = left; i <= right; i++) {
//         if (Number.isInteger(Math.sqrt(i))) {
//             answer -= i;
//         } else {
//             answer += i;
//         }
//     }
//     return answer;
// }